tests/end_to_end.rs

```diff
@@ -1316,3 +1329,173 @@ async fn test_delete_statement() {
+        "|                 | 5555           |                  | 2022-01-01 21:21:21 | 32         |",
+        "|                 | 5555           |                  | 2022-01-01 21:21:21 | 33         |",
+        "|                 | 3333           |                  | 2022-01-01 20:03:03 | 44         |",
+        "+-----------------+----------------+------------------+---------------------+------------+",
+    ];
+    assert_batches_eq!(expected, &results);
+
+    let results =
+        scan_partition(&context, None, partitions[3].clone(), "test_table").await;
+    let expected = vec![
+        "+-----------------+----------------+------------------+---------------------+------------+",
+        "| some_bool_value | some_int_value | some_other_value | some_time           | some_value |",
+        "+-----------------+----------------+------------------+---------------------+------------+",
+        "|                 | 5555           |                  | 2022-01-01 21:21:21 | 32         |",
+        "|                 | 5555           |                  | 2022-01-01 21:21:21 | 31         |",
+        "|                 |                |                  |                     | 40         |",
+        "+-----------------+----------------+------------------+---------------------+------------+"
+    ];
+    assert_batches_eq!(expected, &results);
+
+    //
+    // Execute UPDATE that doesn't change anything
+    //
+    let plan = context
+        .plan_query("UPDATE test_table SET some_bool_value = TRUE WHERE some_value = 200")
+        .await
+        .unwrap();
+    context.collect(plan).await.unwrap();
+
+    assert_partition_ids(&context, 7, vec![2, 3, 5, 6]).await;
+
+    //
+    // Execute UPDATE that references a nonexistent column in the assignment or in the selection,
+    // or results in a type mismatch
+    //
+    let err = context
+        .plan_query("UPDATE test_table SET nonexistent = 42 WHERE some_value = 32")
+        .await
+        .unwrap_err();
+
+    assert!(err
+        .to_string()
+        .contains("Schema error: No field named 'nonexistent'"));
+
+    let err = context
+        .plan_query("UPDATE test_table SET some_value = 42 WHERE nonexistent = 32")
+        .await
+        .unwrap_err();
+
+    assert!(err
+        .to_string()
+        .contains("Schema error: No field named 'nonexistent'"));
+
+    let err = context
+        .plan_query("UPDATE test_table SET some_int_value = 'nope'")
+        .await
+        .unwrap_err();
+
+    assert!(err
+        .to_string()
+        .contains("Cannot cast string 'nope' to value of Int64 type"));
+
+    // This one's a bit different
+    let err = context
+        .plan_query("UPDATE test_table SET some_other_value = 'nope'")
+        .await
+        .unwrap_err();
+
+    assert!(err
+        .to_string()
+        .contains("Unsupported CAST from Utf8 to Decimal128(38, 10)"));
```

Comment:
Is it possible to break these out into a separate test (e.g. `test_update_errors`) just for the sake of higher test precision?

(though OTOH we do test that we can perform an update after an error, so it's worth keeping one error case here)